lib/src/json_schema/validator.dart

```diff
@@ -450,11 +493,21 @@ class Validator {
     /// If the [JsonSchema] is a bool, always return this value.
     if (schema.schemaBool != null) {
       if (schema.schemaBool == false) {
-        _err('${schema.path}: schema is a boolean == false, this schema will never validate. Instance: $instance');
+        _err('schema is a boolean == false, this schema will never validate. Instance: $instance', instance.path,
+            schema.path);
       }
       return;
     }
 
+    if (instance is! Instance) {
```

Comment:
It seems like this might cover over cases where we haven’t properly captured the path for error messages? Is that the case?

Reply:
Not sure how that would happen, can you expound a bit, or think of a test case that would make that happen?

To explain the approach here, `_validate` is called by the public `validate` method with an unwrapped list or map; the first time `Instance` is created we should be at the root of the data structure. Then it's called recursively by the object & list validation methods, which create new `Instance`s with updated paths before calling `_validate`.